### ISSUE TITLE
Error installing sphinx-doc on OS X 10.14 mojave

### ISSUE THREAD

**Issue by sprynmr** [Homebrew/homebrew-core#30195]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Trying to `brew install python@2` on Mac OS X 10.14 mojave, narrowed issue down to installing `brew install sphinx-doc`
- What happened (include command output)
```
Last 15 lines from /Users/me/Library/Logs/Homebrew/sphinx-doc/23.python:
python
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/usr/local/Cellar/sphinx-doc/1.7.6/libexec
--single-version-externally-managed
--record=installed.txt

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
error in Sphinx setup command: Invalid environment marker: python_version<"3.5"
```
- What you expected to happen
Successful installation
- Step-by-step reproduction instructions (by running `brew install` commands)
`brew install sphinx-doc` on mojave

Apologies if this is the wrong place to put this. The gist step is failing due to missing privileges on my GitHub credentials, but I'll try and follow up with that.

**Comment by sfdye** [comment]
I am seeing this as well

**Comment by sprynmr** [comment]
gist-logs:
https://gist.github.com/3f7cf54f72cb566d43f429a6d212f3c6

**Comment by mistydemeo** [comment]
Hmm, interesting. I can't repro using the latest beta and CLT.

**Comment by sprynmr** [comment]
I tested today with the latest and still the same. Curious. Looks like sphinx-doc is using deprecated distutils instead of setup tools, and then it's using the system version of python to execute that, which doesn't recognize the `python_requires` option.

**Comment by DomT4** [comment]
Can't reproduce this one either FWIW, which is unhelpful. I'm running a pure CLT installation, so maybe Xcode is a contributing factor somehow.

**Comment by sprynmr** [comment]
Mind doing a gist-logs on your successful install? Perhaps I can see a difference.

**Comment by sprynmr** [comment]
I got it working. Turns out it wasn't `python_requires`, that was just a warning. It was the `Invalid environment marker: python_version<"3.5"` bit. So I opened up the `setup.py` in the cache folder and just removed the version markers (made assumptions based on the latest python 3 version I'm installing). Then recompressed/zipped, updated the sha256 in my local formula, and installed. 

I'm not sure what the correct solution is though. Found the inspiration here: https://github.com/nicolargo/glances/issues/1064
